pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -611,7 +612,11 @@ private void addNewRow(int docId, GenericRow row) {
         FieldSpec fieldSpec = indexContainer._fieldSpec;
 
         DataType dataType = fieldSpec.getDataType();
-        value = indexContainer._valueAggregator.getInitialAggregatedValue(value);
+        if (indexContainer._isCountValueAggregator) {
```

Comment:
Is this related?

Reply:
yes, related. Please check: `MutableSegmentImplIngestionAggregationTest.testCOUNT` test.

Reply:
I see the reason. The reason is that we don't always replace the column within `COUNT` with `*`. A better way to handle this is to change the `CountValueAggregator` to implement `ValueAggregator<Object, Long>` so that it can accept any object types.

Reply:
Please take a look at the comment above

Reply:
fixed in my local branch. Sending the new iteration soon.